### PR TITLE
fix: use synckit with ESM correctly

### PR DESCRIPTION
**Pull request by JounQin**

### CODE REVIEW

**Comment by JounQin** [comment]
Actually I think this is no need to minify the bundle which makes debug unusable + log useless.

File: package.json [hunk 1]
```diff
@@ -48,11 +48,13 @@
...
     "test": "node ./test/runTest.js",
     "test:update": "SNAPSHOT_UPDATE=true node ./test/runTest.js",
-    "dev": "esbuild ./src/extension.js --bundle --platform=node --target=node16 --external:vscode --external:./worker --outfile=./dist/extension.js",
     "dev:watch": "yarn dev --watch --sourcemap",
-    "build": "yarn dev --minify",
-    "build:worker": "esbuild ./src/worker.js --bundle --platform=node --target=node16 --outfile=./dist/worker.js",
-    "package": "yarn build && yarn build:worker && vsce package --yarn",
+    "build:dev": "yarn build:extension && yarn build:worker",
+    "build": "yarn build:extension --minify && yarn build:worker --minify",
+    "build:extension": "esbuild ./src/extension.js --bundle --platform=node --target=node16 --external:vscode --external:../worker.mjs --outfile=./dist/extension.js",
```
Comment (JounQin) [inline package.json:54]: Maybe we should enable `--sourcemap`.

**Comment by JounQin** [comment]
@idahogurl Any time to review?

File: package.json [hunk 2]
```diff
@@ -98,6 +100,10 @@
       }
     }
   },
+  "vsce": {
+    "dependencies": false,
```
Comment (JounQin) [inline package.json:104]: https://github.com/microsoft/vscode-vsce/issues/914

**Comment by idahogurl** [comment]
> @idahogurl Any time to review?

I will review tomorrow.

**Review by idahogurl — APPROVED** [review]

**Pull request merged**